TELCODOCS:857 - document resource requirements and prerequisites for … (#7)
MirzWeiss/openshift-docs: modules/sandboxed-containers-prereqs-peer-pods.adoc

[id="sandboxed-containers-prerequisites-peer-pods_{context}"]
= Prerequisites

Before you install {sandboxed-containers-first} and enable peer-pods, you must meet the following requirements:

> * You have {product-title} {product-version} installed on AWS or Azure.

Reviewer: Since vSphere is already mentioned, should the above line also include vSphere ?
cc @@jensfr

Author: @bpradipt  Sorry, the section about vSphere should be removed. Not sure why it's still here.

* You have installed the OpenShift CLI (oc).
* You have access to the cluster as a user with the cluster-admin role.

[id="sandboxed-containers-peer-pods-resource-requirements_{context}"]
== Resource requirements for peer-pods in {sandboxed-containers-first}

A peer-pod uses resources in two locations:

> * The Kubernetes worker node. The worker node stores metadata, Kata shim resources (`containerd-shim-kata-v2`), remote-hypervisor (`cloud-api-adaptor`) resources, and the tunnel setup between the worker nodes and the peer-pod VM.

Reviewer: ```suggestion
The Kubernetes worker node. The worker node stores metadata, Kata shim resources (`containerd-shim-kata-v2`), remote-hypervisor (`cloud-api-adaptor`) resources, and the tunnel setup between the worker node and the pod VMs.
```

> * The cloud instance. This is the actual peer-pod VM running in the cloud.

Reviewer: ```suggestion
The cloud instance. This is the actual pod VM created in the cloud in response to pod creation.
```

The CPU and memory resources used in the Kubernetes worker node are handled by the a link:https://kubernetes.io/docs/concepts/scheduling-eviction/pod-overhead/[pod overhead] included in the RuntimeClass (`kata-remote-cc`) definition used for creating peer-pods.

> The total capacity of the peer-pod VMs running in the cloud is defined as Kubernetes Node extended resources. This enables the Kubernetes scheduler to handle capacity tracking and accounting. The extended resource is named `kata.peerpods.io/vm`, and defines the maximum number of peer-pods that you can create in the cluster.

Reviewer: ```suggestion
The total peer-pods capacity of the cluster is the total number of pod VMs that can be created in the cloud and is defined using Kubernetes Node extended resources. This enables the Kubernetes scheduler to handle capacity tracking and accounting. The extended resource is named `kata.peerpods.io/vm`, and defines the maximum number of peer-pods that you can create in the cluster.
```

> A link:https://https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/[mutating webhook] removes any resource-specific entries from the pod specifications and adds them to the peer-pods extended resources. This enables the Kubernetes scheduler to account for these extended resources, ensuring the peer-pod is only scheduled when resources are available.

Reviewer: ```suggestion
A link:https://https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/[mutating webhook] removes any cpu and memory resource-specific entries from the pod specification (if it exists) and adds a peer-pods extended resource to the pod specification. This enables the Kubernetes scheduler to account for these extended resources, ensuring the peer-pod is only scheduled when capacity is available.
```

[id="sandboxed-containers-prerequisites-peer-pods_{context}"]
= Prerequisites

Before you install {sandboxed-containers-first} and enable peer-pods, you must meet the following requirements:

> * You have {product-title} {product-version} installed on AWS.

Author: installed.

* You have installed the OpenShift CLI (oc).

> * You have installed git.

Author: remove

> * You have installed podman.

Author: remove

> * You have installed tar.

Author: remove
